bigquery/google/cloud/bigquery/external_config.py

```diff
@@ -259,47 +376,99 @@ class CSVOptions(object):
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#externalDataConfiguration.csvOptions.allowQuotedNewlines
+        """
+        prop = self._properties.get('allowQuotedNewlines')
+        return prop
 
-    skip_leading_rows = _TypedApiResourceProperty(
-        'skip_leading_rows', 'skipLeadingRows', six.integer_types)
-    """See
-    https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.query.tableDefinitions.(key).csvOptions.skipLeadingRows
-    https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#externalDataConfiguration.csvOptions.skipLeadingRows
-    """
+    @allow_quoted_newlines.setter
+    def allow_quoted_newlines(self, value):
+        self._properties['allowQuotedNewlines'] = value
+
+    @property
+    def encoding(self):
+        """str: The character encoding of the data.
+
+        See
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.query.tableDefinitions.(key).csvOptions.encoding
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#externalDataConfiguration.csvOptions.encoding
+        """
+        prop = self._properties.get('encoding')
+        return prop
+
+    @encoding.setter
+    def encoding(self, value):
+        self._properties['encoding'] = value
+
+    @property
+    def field_delimiter(self):
+        """str: The separator for fields in a CSV file. Defaults a comma (',').
+
+        See
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.query.tableDefinitions.(key).csvOptions.fieldDelimiter
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#externalDataConfiguration.csvOptions.fieldDelimiter
+        """
+        prop = self._properties.get('fieldDelimiter')
+        return prop
+
+    @field_delimiter.setter
+    def field_delimiter(self, value):
+        self._properties['fieldDelimiter'] = value
+
+    @property
+    def quote_character(self):
+        """str: The value that is used to quote data sections in a CSV file.
+
+        See
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.query.tableDefinitions.(key).csvOptions.quote
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#externalDataConfiguration.csvOptions.quote
+        """
+        prop = self._properties.get('quote')
+        return prop
+
+    @quote_character.setter
+    def quote_character(self, value):
+        self._properties['quote'] = value
+
+    @property
+    def skip_leading_rows(self):
+        """int: The number of rows at the top of a CSV file.
+
+        See
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.query.tableDefinitions.(key).csvOptions.skipLeadingRows
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#externalDataConfiguration.csvOptions.skipLeadingRows
+        """
+        prop = self._properties.get('skipLeadingRows')
```

Comment:
We have a function in `_helpers` called `int_or_none` which would be helpful here.